CHANGELOG.md

```diff
@@ -321,17 +321,74 @@ Pragmas and options
   test = refl
   ```
 
-* [**Breaking**] Option `--experimental-lossy-unification` that turns on
-  (the incomplete) first-order unification has been renamed to
-  `--lossy-unification`.
-  ([#1625](https://github.com/agda/agda/issues/1625))
-
-* The new option `--no-load-primitives` complements `--no-import-sorts`
-  by foregoing loading of the primitive modules altogether. This option
-  leaves Agda in a very fragile state, as the built-in sorts are used
-  extensively throughout the implementation. It is intended to be used
-  by Literate Agda projects which want to bind `BUILTIN TYPE` (and
-  other primitives) in their own literate files.
+* [**Breaking**] The option `--experimental-lossy-unification` that
+  makes Agda sometimes use first-order unification has been renamed to
+  `--lossy-unification`
+  ([#1625](https://github.com/agda/agda/issues/1625)).
+
+  Note that with non-lossy unification you have the guarantee (in the
+  absence of bugs) that, if the code type-checks, and you can find one
+  consistent way to instantiate all meta-variables, then that is the
+  way that the code is interpreted by the system. With lossy
+  unification the solution you have in mind might not be the one the
+  system uses.
+
+  Consider the following code, which is based on an example from López
+  Juan's [PhD thesis](https://research.chalmers.se/publication/527051)
+  (see Listing 6.16):
+  ```agda
+  {-# OPTIONS --lossy-unification #-}
+
+  open import Agda.Builtin.Bool
+  open import Agda.Builtin.Nat
+
+  private variable
+    m n : Nat
+
+  infixr 5 _∷_ _++_
+
+  data Bit-vector : Nat → Set where
+    []  : Bit-vector 0
+    _∷_ : Bool → Bit-vector n → Bit-vector (suc n)
+
+  _++_ : Bit-vector m → Bit-vector n → Bit-vector (m + n)
+  []       ++ ys = ys
+  (x ∷ xs) ++ ys = x ∷ (xs ++ ys)
+
+  replicate : ∀ n → Bool → Bit-vector n
+  replicate zero    x = []
+  replicate (suc n) x = x ∷ replicate n x
+
+  vector : Bit-vector (m + n)
+  vector {m = m} {n = n} = replicate m true ++ replicate n false
+  ```
+  Can you confidently predict the values of all of the following four
+  bit-vectors? Are you sure that readers of your code can do this?
+  ```agda
+  ex₁ : Bit-vector (0 + 1)
+  ex₁ = vector
+
+  ex₂ : Bit-vector (1 + 0)
+  ex₂ = vector
+
+  ex₃ : Bit-vector ((0 + 1) + (1 + 0))
+  ex₃ = xs ++ xs
+    where
+    xs = vector
+
+  ex₄ : Bit-vector ((0 + 1) + (1 + 0))
+  ex₄ = xs ++ xs
+    where
+    xs = vector {m = _}
+  ```
```

Comment:
This long disclaimer & example should go to the user manual, with a link from the changelog...

Reply:
We could certainly add something like this to the user manual. However, I think there should be a warning about this option also in the changelog: I don't want to encourage anyone to use it.

Reply:
I think it is good to have a warning in the changelog, but refer to details in the docs.  In the current state, I feel that the discussion of `--lossy-unification` takes over-proportionally much space in the changelog.
What do others think? @jespercockx @plt-amy

Reply:
Having the big explanation directly in the changelog seems a bit excessive, to me... IMO, the warning should be limited to a short paragraph (with a bold **warning** to draw attention), with a sentence or two explaining in generic terms what can go wrong, with a link to further explanation(s) in the user manual.

Reply:
Yes, I agree it's better to have a short entry in the changelog that points to the full explanation in the manual

Reply:
I added some text to the user manual and linked to it.

@asr, the link points to `https://agda.readthedocs.io/en/latest/`. Will you update this link as part of the release process?

Reply:
> @asr, the link points to `https://agda.readthedocs.io/en/latest/`. Will you update this link as part of the release process?

Yes.